src/Tests/Nop.Tests/Nop.Services.Tests/Affiliates/AffiliateServiceTests.cs

```diff
@@ -100,21 +115,48 @@ public async Task CanGetAllAffiliates()
 
             var address = await _addressService.GetAddressByIdAsync(1);
 
-            affiliates = await _affiliateService.GetAllAffiliatesAsync(firstName: address.FirstName ,showHidden: true);
+            affiliates = await _affiliateService.GetAllAffiliatesAsync(firstName: address.FirstName, showHidden: true);
             affiliates.TotalCount.Should().Be(3);
-            affiliates = await _affiliateService.GetAllAffiliatesAsync(firstName: address.FirstName);
+            affiliates = await _affiliateService.GetAllAffiliatesAsync(firstName: address.FirstName, lastName: address.LastName);
             affiliates.TotalCount.Should().Be(2);
+            affiliates = await _affiliateService.GetAllAffiliatesAsync(loadOnlyWithOrders: true, ordersCreatedFromUtc: address.CreatedOnUtc, ordersCreatedToUtc: It.IsAny<DateTime>());
+            affiliates.TotalCount.Should().Be(0);
             affiliates = await _affiliateService.GetAllAffiliatesAsync(loadOnlyWithOrders:true, showHidden: true);
             affiliates.TotalCount.Should().Be(0);
         }
 
+        [Test]
+        public async Task CanUpdateAffiliateAsync()
+        {
+            var newlyAffiliate = new Affiliate() { AddressId = 1 };
+
+            await _affiliateService.InsertAffiliateAsync(newlyAffiliate);
+
+            newlyAffiliate.AdminComment = _activeAffiliate1.AdminComment;
+            newlyAffiliate.FriendlyUrlName = _activeAffiliate1.FriendlyUrlName;
+
+            await _affiliateService.UpdateAffiliateAsync(newlyAffiliate);
+
```

Comment:
This place should contain the code for loading an object from the database, otherwise the check will give nothing, since you assigned new data a couple of lines above